features/publishable/publishable.feature

```diff
@@ -8,231 +8,258 @@ Feature: Access to unpublished/draft resources should be configurable
-    Then the response should include the published resources only
+    Then the response status code should be 200
+    And the response should include the published resources only
 
-  @createSchema
   @loginUser
+  @try
   Scenario: As a user with no draft access, when I get a collection of published resources with draft resources available, and published=false query filter, it should not include the draft resources.
-    Given there are draft and published resources available
+    Given there are 2 draft and published resources available
     When I send a "GET" request to "/component/publishable_components?published=false"
-    Then the response should include the published resources only
+    Then the response status code should be 200
+    And the response should include the published resources only
 
   # POST
-  @createSchema
+  @loginAdmin
+  Scenario: As a user with draft access, when I create a resource with publishedAt=null, I should be able to set the publishedAt date to specify if it is draft/published
+    When I send a "POST" request to "/component/publishable_components" with data:
+      | reference | publishedAt |
+      | test      | null        |
+    Then the response status code should be 201
+    # publishedAt does not exist because it's null
+    And the JSON node publishedAt should not exist
+    And the JSON node published should be false
+
   @loginAdmin
   Scenario Outline: As a user with draft access, when I create a resource, I should be able to set the publishedAt date to specify if it is draft/published
-    When I send a "POST" request to "/component/publishable_components" with body:
-    """
-    {
-      "reference": "test",
-      "publishedAt": "<publishedAt>"
-    }
-    """
-    Then the response should include the key "publishedAt" with the value "<publishedAt>"
+    When I send a "POST" request to "/component/publishable_components" with data:
+      | reference | publishedAt   |
+      | test      | <publishedAt> |
+    Then the response status code should be 201
+    And the response should include the key "publishedAt" with the value "<publishedAt>"
     Examples:
-      | publishedAt         |
-      | null                |
-      | now                 |
-      | 1970-01-01 00:00:00 |
-      | 2999-12-31 23:59:59 |
+      | publishedAt               |
+      | now                       |
+      | 1970-01-01T00:00:00+00:00 |
+      | 2999-12-31T23:59:59+00:00 |
+
+  @loginUser
+  Scenario: As a user with no draft access, when I create a resource with publishedAt=null, I should have the published resource returned, and the publication date is automatically set.
+    When I send a "POST" request to "/component/publishable_components" with data:
+      | reference | publishedAt |
+      | test      | null        |
+    Then the response status code should be 201
+    And the JSON node publishedAt should exist
+    And the JSON node published should be true
 
-  @createSchema
   @loginUser
+  @saveNow
```

Comment:
Behat tests are still failing because of `now` initialized too early :/

Reply:
I'll see if I can look, if not I'll still merge and we can fix on the soft validation PR